#### Multiple Regressiion -- This is similar to linear regression but many values are plotted against the independent variable in multiple regression. More like trying to predict a value with more than one variable.

##### The weight of an engine can be used to predict the amount of CO2 emissions 

#### In this section, from the sklearn module we will use the linearRegression() metthod to create to create a linear regression object.

#### The linearregression() object has a method called fit(). This takes the dependent and independent variables as parameters and fills the linear regression objet with this Data.

In [10]:
import pandas as pd
df = pd.read_csv("Cars.csv")
print(df.head(10))

          Car       Model  Volume  Weight  CO2
0      Toyoty        Aygo    1000     790   99
1  Mitsubishi  Space Star    1200    1160   95
2       Skoda      Citigo    1000     929   95
3        Fiat         500     900     865   90
4        Mini      Cooper    1500    1140  105
5          VW         Up!    1000     929  105
6       Skoda       Fabia    1400    1109   90
7    Mercedes     A-Class    1500    1365   92
8        Ford      Fiesta    1500    1112   98
9        Audi          A1    1600    1150   99


In [9]:
#open data using pandas
import pandas as pd
from sklearn import linear_model
import matplotlib.pyplot as plt 

df = pd.read_csv("Cars.csv")

#index the x and y parameters from your Data frame
x = df[["Weight", "Volume"]]
y = df["CO2"]


regr =  linear_model.LinearRegression()
regr.fit(x, y)

#predict the CO2 emission of a Car where the weight is 2,300kg and the volume is 1,300cm3
predictedCO2 = regr.predict([[2300, 1300]])
print(predictedCO2)

[107.2087328]


##### from the above result we have predicted that a car of 1.3 litre and an engine of 2,300 kilogram will emit 107.2 grams of CO2 every kilometer it travels.

## Coefficient

#### if x is a variable, for 2x, 2 will be the coefficient wile x will be the unknown variable. Coefficient tells us wether or not there is a realtionship between the dependent and independent variables. it tells if a dependent variables changes when an independent variabe is changed.

In [15]:
import pandas as pd
from sklearn import linear_model

df = pd.read_csv("Cars.csv")

x = df[["Weight", "Volume"]]
y = df["CO2"]

regr = linear_model.LinearRegression([[]])
regr.fit(x, y)

print(regr.coef_)

[0.00755095 0.00780526]


#### from the result above it means that when:
#### weight increases by 1kg there will be an emission of 0.00755095
#### volume increases by 1cm3 there will an emission of 0.00780526

In [10]:
def add(a,b):
    add =  a + b;
    print(add)

In [12]:
add(2,3)

5


In [14]:
wel = "hello grant!!"
print(wel)

hello grant!!


In [31]:
def welMess(name):
    print("Hello", name)

In [33]:
welMess("grant")

TypeError: welMess() missing 1 required positional argument: 'name'